### DISCUSSION TITLE
Browser dead with multiple editor render in the same time

### DISCUSSION THREAD

**Discussion by 4-1-1** [Milkdown/milkdown#600] [Q&A] [answered]
We have a  page about test case.
It could be wrote many steps  of the test case.
And each of them is a milkdown editor. 
I have render them with virtualize list. But if the step content is empty . the editor height could be small .
In worst .It will render about 30 steps . means 30 editors.
and then my browser will dead 😮‍💨 

Do not know how to do next.  
Does anyone  had this problem . Help me please 🆘

**Comment by z-950** [comment, marked as answer]
I check the creation flow. The slowest action is to create EditorView, which is handled by prosemirror.

If you only need the html result, try [`getHtml()`](https://milkdown.dev/macros#gethtml).

I think you can show result default. And create editor manually.

**Reply by 4-1-1** [reply]
I am trying to share the DOMSerializer, just use `getHtml()`.
But there is a problem, the DOMSerializer just do serializer by the nodeSpec's toDom. not support NodeView. (react component)
So, It's not work

**Reply by z-950** [reply]
If you accept two steps rendering, I think you can get the react component as a common node with data attritubes first. And then render the react component and replace the mark node.

**Reply by 4-1-1** [reply]
> If you accept two steps rendering, I think you can get the react component as a common node with data attritubes first. And then render the react component and replace the mark node.

Some nodeview is hard to do simple rendering. like xmind.(the saving data is just some json)
So I do a check before render. If has the nodeView node . then do a whole rendering. otherwise to use DOMSerializer

**Reply by z-950** [reply]
> If you accept two steps rendering, I think you can get the react component as a common node with data attritubes first. And then render the react component and replace the mark node.

I think huge data for node can be solved by storing them in another place while leaving an id.
This may can make render smoothly.

**Reply by 4-1-1** [reply]
This way will cost too much storage. not worth it
